arch/Kconfig

```diff
@@ -122,7 +122,7 @@ config RISCV
 	select ARCH_HAS_DIRECTED_IPIS
 	select BARRIER_OPERATIONS_BUILTIN
 	select ARCH_HAS_THREAD_PRIV_STACK_SPACE_GET if USERSPACE
-	imply XIP
+	imply XIP if !SOC_FAMILY_ESPRESSIF_ESP32
```

Comment:
I don't think this should be here. A particular SoC and its configs are built on top of architecture code, so they should not "leak" backwards.

Reply:
@dcpleung `imply` XIP will force XIP no matter what. Do you recommend a way to disable it?

Reply:
Have you tried disabling it in `_defconfig`? That should act as an overlay which should override the default in kconfig files.

Reply:
@wmrsouza Would you check that?

Reply:
I'll check it.

Reply:
I guess the proper way to address this is to push `imply XIP` config into SoC configs and out of the arch layer.

Reply:
> I guess the proper way to address this is to push `imply XIP` config into SoC configs and out of the arch layer.

This the same approach used by xtensa arch.

Reply:
@dcpleung do you thing this approach is valid? (PR was updated moving `imply XIP` to SoCs).

Reply:
I think this is better as the arch should not dictate this kind of SoC/board specific configs.